airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -98,6 +98,15 @@ def __init__(
         log_id_template: str | None = None,
     ):
         es_kwargs = es_kwargs or {}
+        # For elasticsearch>8,arguments like retry_timeout have changed for elasticsearch to retry_on_timeout
+        # in Elasticsearch() compared to previous versions.
+        # Read more at: https://elasticsearch-py.readthedocs.io/en/v8.8.2/api.html#module-elasticsearch
+        if es_kwargs:
+            retry_timeout = es_kwargs.get("retry_timeout")
```

Comment:
IMO, https://github.com/apache/airflow/pull/33135 should be marked as a breaking change. This is one argument that we have found, but there could be more such arguments and adding back compat for them will be challenging and will be discovered only when users face issues based on the params they are using.

Reply:
cc: @Owen-CH-Leung

Reply:
I think it indeed HAS potential of breaking things. If we can make it "reasonably compatible" - i.e. fix back-compatibilities that we know and smooth the migration, that would be great, but I would be for marking the next ES release as MAJOR cc: @eladkal 

Also there is a little twist to it. Unless I am mistaken, I think elasticsearch handler integration was anyhow broken and not working before :) . So well. it could also be seen as bugfix :P

Reply:
I too think we do a major release since the underline dependency had also changed from 7.* to 8.*

Reply:
Okay.

I am not aware of how far it was broken. But, we have few tests and users for whom the elasticsearch handler integration was known to be working well. However, only yesterday our tests caught that the PR broke existing working setup :) 

Perhaps our QA expert @vatsrahul1001 can provide more confirmation here.

Reply:
https://github.com/apache/airflow/pull/33135 did not break any airflow code.

We agreed that bumping dependency is not a breaking change and as expected every time you bump x.y.z to x+1.y.z something can be broken for users.

I am not convencied next release should be a major one.

Reply:
@eladkal , I think if our helm chart can't work with the new Es provider then it should be considered as breaking change. 
Also, maybe we should consider bumping to major versions whenever we bump dependencies to major versions?

Reply:
@eladkal I'm not entirely sure, but the below code was part of airflow's codebase which let to the issue because of upgrade - https://github.com/apache/airflow/blob/01a6c1e7f8b0ed736d7c6f661e0c29ec023016b2/chart/values.yaml#L2149

Reply:
yes, I agree bumping something from x.y.z to x+1.y.z may not need to be a breaking change in general. 👍🏽 

However, would like to bring up and discuss the following point. 
For this provider, elasticsearch is the main underlying dependency, and we're updating it to the next major version from 7.x to 8.x. It does not affect core Airflow code/functionality, but since providers are versioned and released independently, upgrading to this provider release might affect existing deployments for users.

Reply:
@ephraimbuddy I didn't see helm tests were broken?
and to my understanding this PR adds support for elastic search 8 so what is the motivation for the breaking change?

> Also, maybe we should consider bumping to major versions whenever we bump dependencies to major versions?

This was discussed and had lazy consequence if I remember correctly. Need to lookup the thread.